spyder/plugins/console/widgets/main_widget.py

```diff
@@ -234,6 +234,8 @@ def setup(self, options):
             ConsoleWidgetMenus.InternalSettings,
             _('Internal console settings'),
         )
+        internal_settings_menu.menuAction().setIconVisibleInMenu(True)
+        internal_settings_menu.setIcon(self.create_icon('tooloptions'))
```

Comment:
@goanpeca shouldn't this setup be supported by the `create_menu` method? Something like:
```python
self.create_menu(..., icon=self.create_icon('tooloptions'))
```

Reply:
Maybe 🤷🏻‍♂️ want to open a new issue to add that to the new API?

Reply:
Sure :+1:

Reply:
@goanpeca you want to address #13695 here or in another PR?

Reply:
Other PR.

Reply:
Ok :+1: please open the other PR for #13695 and after that one is merged we can merge this one

Reply:
https://github.com/spyder-ide/spyder/pull/13709

Reply:
@goanpeca could you update this one to reflect the changes in the API made in #13709? Thanks :+1:

Reply:
Done @dalthviz.